src/main/scala/viper/gobra/frontend/info/implementation/resolution/MemberResolution.scala

```diff
@@ -241,18 +248,65 @@ trait MemberResolution { this: TypeInfoImpl =>
   lazy val tryUnqualifiedPackageLookup: PIdnUse => Entity =
     attr[PIdnUse, Entity] {
       id => {
-        val unqualifiedImports = tree.root.imports.collect { case ui: PUnqualifiedImport => ui }
-        val results = unqualifiedImports.map(ui => tryPackageLookup(ui, id)).collect { case Some(r) => r }
-        if (results.isEmpty) {
-          UnknownEntity()
-        } else if (results.length > 1) {
-          MultipleEntity()
-        } else {
-          results.head._1
+        val transitiveParent = transitiveClosure(id, tree.parent(_))
+        val entities = for {
+          // get enclosing PProgram for this PIdnUse node
+          program <- transitiveParent(id).collectFirst { case p: PProgram => p }
+          // consider all unqualified imports for this program (not package)
+          unqualifiedImports = program.imports.collect { case ui: PUnqualifiedImport => ui }
+          // perform a package lookup in each unqualifiedly imported package
+          results = unqualifiedImports.map(ui => tryPackageLookup(ui, id)).collect { case Some(r) => r }
```

Comment:
Can you not just use faltMap and then not do the collect{ case Some(r) => r } part afterwards?

Reply:
you're right